src/main/java/edu/ie3/netpad/io/controller/IoDialogs.java

```diff
@@ -57,6 +59,135 @@
     return csvFileSeparatorDialog;
   }
 
+  /**
+   * Creates a dialog, that is used to ask the user for details on how the csv data is meant to look
+   * like in detail.
+   *
+   * @param title Window title
+   * @param directoryButtonText Button text for the directory button
+   * @param archiveButtonText Button text for the archive button
+   * @return Detailed information about the csv details as {@link CsvIoData}
+   */
+  public static Dialog<CsvIoData> csvIoDialog(
+      String title, String directoryButtonText, String archiveButtonText) {
+    GridPane gridPane = new GridPane();
+    gridPane.setHgap(10);
+    gridPane.setVgap(10);
+    gridPane.setPadding(new Insets(20, 150, 10, 10));
+
+    /* Ask for the csv separator */
+    Label separatorLbl = new Label(".csv file separator: ");
+    final ComboBox<String> separatorCb =
+        new ComboBox<>(FXCollections.observableArrayList(";", ","));
+    separatorCb.getSelectionModel().selectFirst();
+    gridPane.addRow(0, separatorLbl, separatorCb);
+
+    /* Select the directory hierarchy */
+    Label hierarchyLbl = new Label("Directory hierarchy: ");
+    ToggleGroup tglGrp = new ToggleGroup();
+    ToggleButton flatBtn = new RadioButton("flat");
+    flatBtn.setUserData(CsvIoData.DirectoryHierarchy.FLAT);
+    flatBtn.setToggleGroup(tglGrp);
+    ToggleButton hierarchicBtn = new RadioButton("hierarchic");
+    hierarchicBtn.setUserData(CsvIoData.DirectoryHierarchy.HIERARCHIC);
+    hierarchicBtn.setToggleGroup(tglGrp);
+    tglGrp.selectToggle(flatBtn);
+    gridPane.addRow(1, hierarchyLbl, flatBtn, hierarchicBtn);
+
+    DialogPane dialogPane = new DialogPane();
+    dialogPane.setContent(gridPane);
+
+    ButtonType directoryButtonType = new ButtonType(directoryButtonText);
+    ButtonType archiveButtonType = new ButtonType(archiveButtonText);
+
+    dialogPane.getButtonTypes().addAll(directoryButtonType, archiveButtonType, ButtonType.CANCEL);
+
+    Dialog<CsvIoData> csvImportDialog = new Dialog<>();
+    csvImportDialog.setTitle(title);
+    csvImportDialog.setDialogPane(dialogPane);
+
+    csvImportDialog.setResultConverter(
+        buttonType -> {
+          String csvSeparator = separatorCb.getSelectionModel().getSelectedItem();
+          CsvIoData.DirectoryHierarchy hierarchy =
+              (CsvIoData.DirectoryHierarchy) tglGrp.getSelectedToggle().getUserData();
+          if (buttonType.equals(directoryButtonType)) {
+            return new CsvIoData(csvSeparator, hierarchy, CsvIoData.SourceType.DIRECTORY);
+          } else if (buttonType.equals(archiveButtonType)) {
+            return new CsvIoData(csvSeparator, hierarchy, CsvIoData.SourceType.ARCHIVE);
+          } else {
+            return null;
```

Comment:
is there a need for another `else` and returning `null` here? If possible pls consider 

```java 
 if (buttonType.equals(directoryButtonType)) {
            return new CsvIoData(csvSeparator, hierarchy, CsvIoData.SourceType.DIRECTORY);
          } else {
            return new CsvIoData(csvSeparator, hierarchy, CsvIoData.SourceType.ARCHIVE);
          }  
```

You might make it even more "bulletproof" by throwing an exception for every other type of button that is not known e.g.

```java
  if (buttonType.equals(directoryButtonType)) {
            return new CsvIoData(csvSeparator, hierarchy, CsvIoData.SourceType.DIRECTORY);
          } else if (buttonType.equals(archiveButtonType)) {
            return new CsvIoData(csvSeparator, hierarchy, CsvIoData.SourceType.ARCHIVE);
          } else {
            throw new NetPadPlusPlusException("Invalid button type "+ buttonType +" in CsvImportDialog")
         }
```